File: Backend/ice-service/dto/ice_dto.go
package dto

type CreateIceCreamRequest struct {
	IceCreamName   string `json:"ice_cream_name"`
	IceCreamFlavor string `json:"ice_cream_flavor"`
	IceCreamPrice  int    `json:"ice_cream_price"`
	IceCreamDesc   string `json:"ice_cream_desc"`
	ImageURL       string `json:"image_url"`
}

type DeleteIceCreamRequest struct {
	IceCreamID string `json:"ice_cream_id"`
}

type UpdateIceCreamRequest struct {
	IceCreamID     string `json:"ice_cream_id"`
	IceCreamName   string `json:"ice_cream_name"`
	IceCreamFlavor string `json:"ice_cream_flavor"`
	IceCreamPrice  int    `json:"ice_cream_price"`
	IceCreamDesc   string `json:"ice_cream_desc"`
	ImageURL       string `json:"image_url"`
}

type MessageResponse struct {
	Message string `json:"message"`
}

type IceCreamResponse struct {
	IceCreamID     string `json:"ice_cream_id"`
	IceCreamName   string `json:"ice_cream_name"`
	IceCreamFlavor string `json:"ice_cream_flavor"`
	IceCreamPrice  int    `json:"ice_cream_price"`
	IceCreamDesc   string `json:"ice_cream_desc"`
	ImageURL       string `json:"image_url"`
}

type ListIceCreamRequest struct {
	Keyword string `form:"keyword"`
	Page    int    `form:"page"`
	Limit   int    `form:"limit"`
}

type ListIceCreamResponse struct {
	Items []IceCreamResponse `json:"items"`
	Total int64              `json:"total"`
}

File: frontend/src/dto/iceDto.ts
export interface CreateIceCreamRequest {
	ice_cream_name   : string
	ice_cream_flavor : string
	ice_cream_price  : number   
	ice_cream_desc   : string
	image_url       : string
}

export interface DeleteIceCreamRequest {
	ice_cream_id : string
}

export interface UpdateIceCreamRequest {
	ice_cream_id     : string
	ice_cream_name   : string
	ice_cream_flavor : string
	ice_cream_price  : number   
	ice_cream_desc   : string
	image_url       : string
}

export interface MessageResponse {
	message : string
}

export interface IceCreamResponse {
	ice_cream_id     : string
	ice_cream_name   : string
	ice_cream_flavor : string
	ice_cream_price  : number   
	ice_cream_desc   : string
	image_url       : string
}

export interface ListIceCreamRequest {
	keyword : string
	page    : number   
	limit   : number   
}

export interface ListIceCreamResponse {
	items : IceCreamResponse[]
	total : number
}
